### PR TITLE
Fix school request

### PR DESCRIPTION
**Pull request by Karonuk**

### CODE REVIEW

**Review by Redingston — DISMISSED** [review]

File: src/app/services/school-requests/school-request.service.ts [hunk 1]
```diff
@@ -2,29 +2,31 @@ import { apiUrl } from 'src/constants/urls';
...
-    return this.http.post<any>(this.url,{id:id},this.httpOptions.options);
+    return this.http.post(this.url+"/Approve",{id:id},this.httpOptions.options).subscribe((res) => {this.get()});
   };
 
-  disapprove = (id:number): Observable<any> => {
+  disapprove = (id:number) => {
     this.httpOptions.loadHeaders();
-    return this.http.delete<any>(this.url+"?id="+id,this.httpOptions.options);
+    return this.http.delete(this.url+"?id="+id,this.httpOptions.options).subscribe((res) => {this.get()});
   };
```
Comment (DmMov) [inline src/app/services/school-requests/school-request.service.ts]: The 28th and 30th lines dont have the spaces they need.

File: src/app/services/school-requests/school-request.service.ts [hunk 2]
```diff
@@ -2,29 +2,31 @@ import { apiUrl } from 'src/constants/urls';
...
+      this.requests=res;
+    } );
   };
 
-  approve = (id:number): Observable<any> => {
+  approve = (id:number) => {
     this.httpOptions.loadHeaders();
-    return this.http.post<any>(this.url,{id:id},this.httpOptions.options);
+    return this.http.post(this.url+"/Approve",{id:id},this.httpOptions.options).subscribe((res) => {this.get()});
   };
```
Comment (DmMov) [inline src/app/services/school-requests/school-request.service.ts]: Spaces.

Comment (Karonuk) [reply src/app/services/school-requests/school-request.service.ts]: each one?

File: src/app/services/school-requests/school-request.service.ts [hunk 3]
```diff
@@ -2,29 +2,31 @@ import { apiUrl } from 'src/constants/urls';
...
+  private url: string = apiUrl + "/SchoolRequests";
+  constructor(private http:HttpClient, private httpOptions:HttpOptionsService) { }
+  requests:ISchoolRequestVM[];
 
-  get = (): Observable<ISchoolRequestVM[]> => {
+  get = () => {
     this.httpOptions.loadHeaders();
-    return this.http.get<ISchoolRequestVM[]>(this.url,this.httpOptions.options);
+    return this.http.get(this.url,this.httpOptions.options).subscribe( (res: ISchoolRequestVM[]) => {
+      this.requests=res;
```
Comment (DmMov) [inline src/app/services/school-requests/school-request.service.ts]: Spaces.
There is needed and not needed spaces.

File: src/app/services/school-requests/school-request.service.ts [hunk 4]
```diff
@@ -2,29 +2,31 @@ import { apiUrl } from 'src/constants/urls';
...
   providedIn: 'root'
 })
-export class SchoolRequestService {
+export class SchoolRequestsService {
 
-  private url: string = apiUrl+"/SchoolRequestProccesing";
-  constructor(private http:HttpClient,private httpOptions:HttpOptionsService) { }
+  private url: string = apiUrl + "/SchoolRequests";
+  constructor(private http:HttpClient, private httpOptions:HttpOptionsService) { }
+  requests:ISchoolRequestVM[];
```
Comment (DmMov) [inline src/app/services/school-requests/school-request.service.ts]: Same

File: src/app/components/School Requests/school-requests/school-requests.component.ts [hunk 5]
```diff
@@ -11,18 +8,11 @@ import { SchoolRequestService } from 'src/app/services/school-requests/school-re
 })
 export class SchoolRequestsComponent implements OnInit {
 
-  Requests:ISchoolRequestVM[];
 
-  constructor(private requestClient:SchoolRequestService) { }
+
+  constructor(public schoolRequestService:SchoolRequestsService) { }
```
Comment (DmMov) [inline src/app/components/School Requests/school-requests/school-requests.component.ts]: There are a few issues.

File: src/app/components/School Requests/school-request-card/school-request-card.component.ts [hunk 6]
```diff
@@ -11,20 +12,17 @@ import { SchoolRequestsComponent } from '../school-requests/school-requests.comp
 })
 export class SchoolRequestCardComponent implements OnInit {
 
-
   @Input() Request:ISchoolRequestVM;
-  constructor(private client:SchoolRequestService,private parent:SchoolRequestsComponent) {  }
+  constructor(private client:SchoolRequestsService) {  }
```
Comment (DmMov) [inline src/app/components/School Requests/school-request-card/school-request-card.component.ts]: Again

File: src/app/components/School Requests/school-request-card/school-request-card.component.html [hunk 7]
```diff
@@ -1,12 +1,22 @@
...
+  <span class="locality-name value">{{Request.locality}}</span>
+</p>
+<p class="address value-wrap">
+  <mat-icon class="icon">place</mat-icon>
+  <span class="address-value value">{{Request.address}}</span>
+</p>
+<div class="button-row">
+  <button class="btn" (click)="approve()">approve</button>
+  <button class="decline btn" (click)="decline()">decline</button>
+</div>
```
Comment (DmMov) [inline src/app/components/School Requests/school-request-card/school-request-card.component.html:22]: Perfect.

File: src/app/components/School Requests/school-request-card/school-request-card.component.scss [hunk 8]
```diff
@@ -1,21 +1,50 @@
...
+      background-color: $red;
+      border-color: $red;
+
+      &:hover {
+        background: none;
+        color: $red;
+      }
+    }
+  }
 }
```
Comment (DmMov) [inline src/app/components/School Requests/school-request-card/school-request-card.component.scss:50]: Perfect.

File: src/app/components/School Requests/school-request-card/school-request-card.component.html (hunk 7 above)
Comment (Karonuk) [reply src/app/components/School Requests/school-request-card/school-request-card.component.html:22]: self-critical

File: src/app/components/School Requests/school-request-card/school-request-card.component.scss (hunk 8 above)
Comment (Karonuk) [reply src/app/components/School Requests/school-request-card/school-request-card.component.scss:50]: self-critical

File: src/app/services/school-requests/school-request.service.ts (hunk 2 above)
Comment (DmMov) [reply src/app/services/school-requests/school-request.service.ts]: You need to add 9 spaces to this piece of code.

**Review by Redingston — DISMISSED** [review]

**Review by Hrabarchuk — APPROVED** [review]

**Review by DmMov — APPROVED** [review]

**Review by Redingston — APPROVED** [review]

**Pull request merged**